### DISCUSSION TITLE
git commands for contributing in Flask

### DISCUSSION THREAD

**Discussion by Yourun-proger** [pallets/flask#4247] [Q&A] [answered]
I do not know if it is possible to ask this type of question here, but I hope that it is possible and that they will help me.
I wanted to contribute to the flask a second time, but wondered about the git.
I have local branches like this:

* fix (my first contribution branch)
* main

now I'm in the **fix** branch
I would like to make a new branch, but what is the correct way to do it?
So:
```bash
git checkout main
git checkout -b my-branch-name origin/2.0.x
```

or like this:
```bash
git checkout main
git fetch origin
git checkout -b my-branch-name origin/2.0.x
```

Or something else? Sorry for the possibly stupid question. I just don't want to mess up my fork and local repo.
I've read the resources for these commands, but I'm not sure how to do...

**Comment by greyli** [comment, marked as answer]
The [contributing guide](https://flask.palletsprojects.com/en/2.0.x/contributing/#start-coding) contains everything you need to know to contribute to Flask.

**Reply by Yourun-proger** [reply]
Thank you so much!
